tree/forest/v7/inc/ROOT/RColumnElement.hxx

```diff
@@ -0,0 +1,96 @@
+/// \file ROOT/RColumnElement.hxx
+/// \ingroup Forest ROOT7
+/// \author Jakob Blomer <jblomer@cern.ch>
+/// \date 2018-10-09
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2015, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RColumnElement
+#define ROOT7_RColumnElement
+
+#include <ROOT/RColumnModel.hxx>
+
+#include <cstring> // for memcpy
+
+namespace ROOT {
+namespace Experimental {
+
+namespace Detail {
+
+// clang-format off
+/**
+\class ROOT::Experimental::Detail::RColumnElement
+\ingroup Forest
+\brief A column element points into a memory mapped page, to a particular data item
+*/
+// clang-format on
+class RColumnElementBase {
+protected:
+   /// Indicates that fRawContent is bitwise identical to the type of the RColumnElement
+   bool fIsMovable;
+   void *fRawContent;
+   unsigned fSize;
+
+   EColumnType fColumnType;
+
+   virtual void DoSerialize(void* /* destination */) const { }
+   virtual void DoDeserialize(void* /* source */) const { }
+
+public:
+   RColumnElementBase()
+     : fIsMovable(false)
+     , fRawContent(nullptr)
+     , fSize(0)
+     , fColumnType(EColumnType::kUnknown) { }
+   virtual ~RColumnElementBase() { }
+
+   void Serialize(void *destination) const {
+     if (!fIsMovable) {
+       DoSerialize(destination);
+       return;
+     }
+     std::memcpy(destination, fRawContent, fSize);
+   }
```

Comment:
It would be better if this `if` could be resolved at compile time, given that the condition doesn't change for a given type. If these functions are moved to `RColumnElement`, then you can use `T` with [`std::is_trivially_copyable`](https://en.cppreference.com/w/cpp/types/is_trivially_copyable), for example, to decide if you can use `memcpy()`.